### PR TITLE
Redo Shoot Command and Add SetShooterWithLimelight Command

### LINKED ISSUE #45: New Shoot Command, Rename Existing Shoot Command
Rename Shoot to SetShooterWithLimelight or something. This command doesn’t actually shoot, it should just sets the wheel speed and hood angle using the limelight. 

Make a new “Shoot” command which checks the keys and runs the revolver only.

### PR DESCRIPTION
**Pull request by Jacob4649**
Fix #45

### CODE REVIEW

File: src/main/java/frc/robot/commands/revolver/Shoot.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import frc.robot.subsystems.Revolver;
+import frc.robot.subsystems.Shooter;
+import frc.robot.subsystems.Turret;
+
+public class Shoot extends CommandBase {
+
+  Revolver m_revolver;
+  Shooter m_shooter;
+  Hood m_hood;
+  Turret m_turret;
```
Comment (matthew-reynolds) [inline src/main/java/frc/robot/commands/revolver/Shoot.java]: Private final

File: src/main/java/frc/robot/commands/revolver/Shoot.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  /**
+   * Starts the revolver and feedwheel if the shooter speed, turret angle, and
+   * hood angle are all at the setpoints
+   * 
+   * 
+   * @param revolver the revolver to use for this command
+   * @param shooter  the revolver to use for this subsystem
+   * @param turret   the revolver to use for this subsystem
+   * @param hood     the revolver to use for this subsystem
```
Comment (matthew-reynolds) [inline src/main/java/frc/robot/commands/revolver/Shoot.java]: Looks like a copy-paste error here

File: src/main/java/frc/robot/commands/revolver/Shoot.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+   */
+  public Shoot(Revolver revolver, Shooter shooter, Turret turret, Hood hood) {
+    m_revolver = revolver;
+    addRequirements(m_revolver);
+  }
+
+  @Override
+  public void initialize() {
+  }
```
Comment (matthew-reynolds) [inline src/main/java/frc/robot/commands/revolver/Shoot.java]: Remove this

File: src/main/java/frc/robot/commands/shooter/SetShooterWithLimelight.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+/**
+ * Instant command to shoot
+ */
+public class SetShooterWithLimelight extends CommandBase {
+
+  private Shooter m_shooter;
+  Turret m_turret;
+  private Hood m_hood;
+  private Limelight m_limelight;
```
Comment (matthew-reynolds) [inline src/main/java/frc/robot/commands/shooter/SetShooterWithLimelight.java]: private final

File: src/main/java/frc/robot/util/Utilities.java [hunk 5]
```diff
@@ -37,4 +37,22 @@ public static double deadzone(double input, double deadzone) {
...
+  }
+
+  /**
+   * Gets a shooter wheel speed from a distance to the limelight target
+   * @param distance the distance to the limelight target in inches
+   * @return the desired wheel speed in rpm
+   */
+  public static double distanceToSpeed(double distance) {
+    return 0;
+  }
```
Comment (matthew-reynolds) [inline src/main/java/frc/robot/util/Utilities.java]: These don't belong in `Utilities.java`.

The `Utilities` class is just for generic math type things, this logic should probably be in the `Limelight` object.

File: src/main/java/frc/robot/subsystems/Hood.java [hunk 6]
```diff
@@ -47,6 +47,14 @@ public double getMeasurement() {
     return getAngle();
   }
 
+  /**
+   * Gets whether the hood is at the desired angle
+   * @return true if the hood is at the desired angle
```
Comment (matthew-reynolds) [inline src/main/java/frc/robot/subsystems/Hood.java]: Nitpick: You don't need this duplication for simple javadocs

File: src/main/java/frc/robot/commands/shooter/SetShooterWithLimelight.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Hood;
+import frc.robot.subsystems.Limelight;
+import frc.robot.subsystems.Shooter;
+import frc.robot.subsystems.Turret;
+import frc.robot.util.Utilities;
+
+/**
+ * Instant command to shoot
```
Comment (matthew-reynolds) [inline src/main/java/frc/robot/commands/shooter/SetShooterWithLimelight.java]: Why is this instant, don't we need to keep doing this while the button is held to keep updating the setpoints?

**Review by matthew-reynolds — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/shooter/SetShooterWithLimelight.java (hunk 7 above)
Comment (Jacob4649) [reply src/main/java/frc/robot/commands/shooter/SetShooterWithLimelight.java]: resolved in d438c31

File: src/main/java/frc/robot/commands/shooter/SetShooterWithLimelight.java (hunk 4 above)
Comment (Jacob4649) [reply src/main/java/frc/robot/commands/shooter/SetShooterWithLimelight.java]: resolved in d438c31

File: src/main/java/frc/robot/commands/revolver/Shoot.java (hunk 3 above)
Comment (Jacob4649) [reply src/main/java/frc/robot/commands/revolver/Shoot.java]: resolved in 3a5a41b

File: src/main/java/frc/robot/commands/revolver/Shoot.java (hunk 2 above)
Comment (Jacob4649) [reply src/main/java/frc/robot/commands/revolver/Shoot.java]: resolved in 3a5a41b

File: src/main/java/frc/robot/commands/revolver/Shoot.java (hunk 1 above)
Comment (Jacob4649) [reply src/main/java/frc/robot/commands/revolver/Shoot.java]: resolved in 3a5a41b

File: src/main/java/frc/robot/subsystems/Hood.java (hunk 6 above)
Comment (Jacob4649) [reply src/main/java/frc/robot/subsystems/Hood.java]: resolved in 1e533bc

File: src/main/java/frc/robot/util/Utilities.java (hunk 5 above)
Comment (Jacob4649) [reply src/main/java/frc/robot/util/Utilities.java]: resolved in 4bfc2e7

File: src/main/java/frc/robot/commands/revolver/Shoot.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+  @Override
+  public void end(boolean interrupted) {
+    m_revolver.stop();
+  }
+
+  @Override
+  public boolean isFinished() {
+    return false;
+  }
```
Comment (matthew-reynolds) [inline src/main/java/frc/robot/commands/revolver/Shoot.java]: Remove

File: src/main/java/frc/robot/commands/revolver/Shoot.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+  private final Revolver m_revolver;
+  private final Shooter m_shooter;
+  private final Hood m_hood;
+  private final Turret m_turret;
+
+  /**
+   * Starts the revolver and feedwheel if the shooter speed, turret angle, and
+   * hood angle are all at the setpoints
+   * 
+   *
```
Comment (matthew-reynolds) [inline src/main/java/frc/robot/commands/revolver/Shoot.java]: Double newline

**Review by matthew-reynolds — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/revolver/Shoot.java (hunk 9 above)
Comment (Jacob4649) [reply src/main/java/frc/robot/commands/revolver/Shoot.java]: resolve in f471943

File: src/main/java/frc/robot/commands/revolver/Shoot.java (hunk 8 above)
Comment (Jacob4649) [reply src/main/java/frc/robot/commands/revolver/Shoot.java]: resolve in f471943

**Review by matthew-reynolds — APPROVED** [review]

**Pull request merged**